ajax/ajaxVote.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+include('config/init.php');
+
+if(!isset($_SESSION['userID'])) {
+	echoNotLoggedIn("You must be logged in to vote.");
+	die();
+}
+
+$eventID=$_REQUEST['eventID'];
+$sessionUserID=$_SESSION['userID'];
+$direction=$_REQUEST['direction'];
+
+$userVote = doesUserVoteExist($eventID, $sessionUserID);
+
+
+//also assumed 3 cases: 1) insert&vote 2)delete&vote and 3)delete, insert, and vote
+if($userVote == null){
+	insertUserVote($eventID, $sessionUserID, $direction);
+	generalVoteinDB($eventID, $direction);
+}
+
+
+else if(($userVote['0']['direction'] == "down" && $direction == "down")||($userVote['0']['direction'] == "up" && $direction == "down")){
+	deleteUserVote($userVote['0']['eventID'], $sessionUserID);
```

Comment:
This works, but I'd just handle undos separate from this whole if clause. At the top of this file, just see if the user has already voted on this event, and if they have deleted it. Then, just insert the new vote (without regard for whether an undo just happened). Separating them out will simplify this.